File: apps/angular/src/app/models/Plan.ts
export class Plan {
    clean:boolean;
    code:string;
    createdOn:string;
    id: string;
    locked:boolean;
    valid: string;
    params: string;
    version: string;
    constructor(p: any) {
        this.id = p.id
        this.code = p.code
        this.createdOn = p.createdOn
        this.valid = p.valid 
        this.locked = p.locked 
        this.params = p.params
        this.version = p.version
        this.clean = p.clean
    }
}
 

File: apps/metaheuristic-api/src/main/java/ai/metaheuristic/api/launchpad/Plan.java
/*
 * Metaheuristic, Copyright (C) 2017-2019  Serge Maslyukov
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, version 3 of the License.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package ai.metaheuristic.api.launchpad;

/**
 * @author Serge
 * Date: 5/9/2019
 * Time: 3:27 PM
 */
public interface Plan {
    Long getId();

    Integer getVersion();

    String getCode();

    long getCreatedOn();

    String getParams();

    boolean isLocked();

    boolean isValid();

    boolean isClean();

    void setId(Long id);

    void setVersion(Integer version);

    void setCode(String code);

    void setCreatedOn(long createdOn);

    void setParams(String params);

    void setLocked(boolean locked);

    void setValid(boolean valid);

    void setClean(boolean clean);
}
